src/test/java/updated/game/abstractClasses/Board.java

```diff
@@ -0,0 +1,150 @@
+package updated.game.abstractClasses;
+
+import lombok.Getter;
+
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+@Getter
+public abstract class Board {
+    public static  int BOARD_SIZE = 3;
+    protected List<List<Character>> board;
+    public Set<Integer> emptyCells;
+    public char winningSym;
+
+    public Board(){
+        emptyCells = new LinkedHashSet<>(BOARD_SIZE*BOARD_SIZE);
+        this.initEmptyCells();
+
+    }
+
+
+    public void initEmptyCells(){
+        SecureRandom randomNumGen = new SecureRandom();
+        int size = BOARD_SIZE * BOARD_SIZE;
+        while (this.emptyCells.size() != size) {
+            int i = randomNumGen.nextInt(size+1);
+            if (i == 0)
+                i++;
+            this.emptyCells.add(i);
+        }
+        this.winningSym = ' ';
+
+        if (this.board != null) {
+            this.board.clear();
+        }
+        this.board = new ArrayList<>(BOARD_SIZE);
+        for (int row = 0; row < BOARD_SIZE; row++) {
+            this.board.add(new ArrayList<>(BOARD_SIZE));
+            for (int col = 0; col < BOARD_SIZE; col++) {
+                this.board.get(row).add(' ');
+            }
+        }
+    }
+    public boolean checkCollision(int index){
+        return !this.emptyCells.contains(index);
+    }
+  
+    public void userMove(Gamer gamer, int row, int col){
+        gamer.incrementNumberOfMoves();
+        this.board.get(row).set(col, gamer.getSymbol());
+    }
+
+    public boolean checkDiagonal() {
```

Comment:
The diagonal check logic is a bit hard to follow and specific to a 3x3 grid. To support an n x n grid, you could generalize the diagonal logic for variable sizes.